src/elements/components/FeatheryAuthGate.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Form } from '../..';
+import { Props as FormProps } from '../../Form';
+import { getStytchJwt } from '../../utils/browser';
+import { getAuthClient, initInfo, initState } from '../../utils/init';
+import { authHookCb } from '../../integrations/stytch';
+import Client from '../../utils/client';
+import { isAuthStytch } from '../../integrations/utils';
+import Spinner from './Spinner';
+import { isHrefFirebaseMagicLink } from '../../integrations/firebase';
+/** TODO: These next 2 should maybe be dynamically imported, but having trouble with that
+ * combined 6.9k gzipped, so OK for now
+ */
+import { useIdleTimer } from 'react-idle-timer';
+import throttle from 'lodash.throttle';
+
+const TEN_SECONDS_IN_MILLISECONDS = 1000 * 10;
+const FIVE_MINUTES_IN_MILLISECONDS = 1000 * 60 * 5;
+
+export const authState = {
+  // This is a flag so we only redirect to the login start step immediately
+  // after auth, not during other form navigation
+  redirectAfterLogin: false,
+  sentAuth: false,
+  clearLoader: () => {},
+  onLogin: () => {},
+  onLogout: () => {}
+};
+
+const FeatheryAuthGate = ({
+  authId: authIdProp,
+  formProps,
+  whitelistPath,
+  fullPageLogin,
+  onLogin = () => {},
+  onLogout = () => {},
+  children
+}: {
+  authId?: string;
+  formProps: FormProps;
+  whitelistPath?: string;
+  fullPageLogin?: boolean;
+  onLogin?: () => void;
+  onLogout?: () => void;
+  children: JSX.Element;
+}) => {
+  const formCompleted = initInfo().sessions[formProps.formName]?.form_completed;
```

Comment:
instead of needing to store sessions globally, can't you leverage the onFormComplete callback handler exposed by the Form component?

Reply:
and if that callback handler event is buggy (which it might be), it'd probably be better to robustify the handler rather than store session globally

Reply:
fair point about using the onFormComplete callback, that is probably a more direct way to handle this

Reply:
oh, the reason I didn't do that is because this logic isn't just needed for when the onboarding questions are completed, but also determining if the onboarding form was already completed in the past

there was already an object to store sessions in initState so I leveraged that for this info